src/main/daml/Daml/Finance/Interface/Lifecycle/Rule/NewVersion.daml

```diff
@@ -0,0 +1,55 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Interface.Lifecycle.Rule.NewVersion where
+
+-- | Type synonym for `NewVersion`.
+type I = NewVersion
+
+-- | Type synonym for `View`.
+type V = View
+
+-- | View for `NewVersion`.
+data View = View
+  with
+    lifecycler : Party
+      -- ^ Party performing the lifecycling.
+  deriving (Eq, Show)
+
+-- | Interface implemented by instruments that create a contingent claims tree on-the-fly.
+interface NewVersion where
+  viewtype V
+
+  createNewVersion : CreateNewVersion -> Update (ContractId NewVersion)
+  getLastEventTimestamp : GetLastEventTimestamp -> Update (Time)
+
+  nonconsuming choice GetView : View
+    -- ^ Retrieves the interface view.
+    with
+      viewer : Party
+        -- ^ The party retrieving the view.
+    controller viewer
+    do
+      pure $ view this
+
+  nonconsuming choice GetLastEventTimestamp : Time
```

Comment:
This could be part of the view as time is serialisable

Reply:
I added `lastEventTimestamp` to the view, but how do I extract it in the Rule? I tried a couple of ways, but got different kinds of errors.

Reply:
Either you `exerciseInterfaceByKey GetView` or you `fetchInterfaceByKey` and then call the `view` built-in method.

Reply:
`exerciseInterfaceByKey GetView` worked fine, thanks!